concordia/static/js/action-app/index.js

```diff
@@ -839,6 +882,13 @@ export class ActionApp {
                     continue;
                 }
 
+                if (
+                    currentTopicId &&
+                    !this.assetHasTopic(asset, currentTopicId)
```

Comment:
I'm wondering whether this should simply be `!asset.topics.contains(currentTopicId)`

Reply:
Only if I flatten `asset.topics` to a plain array of just IDs - it's currently an array of dictionaries.

Reply:
Ah, right – I hadn't gotten that far in the code yet. It might be nice to avoid a check-at-a-distance but I'm not sure that's worth changing `topics` into a dictionary.